spotinst/resource_spotinst_ocean_right_sizing_rule.go

```diff
@@ -380,15 +401,82 @@ func expandWorkloads(data interface{}) ([]*right_sizing.Workload, error) {
 			workload.RegexName = spotinst.String(v)
 		}
 
-		workloads = append(workloads, workload)
+		//During first workload API call responseDataWorkload, responseDataRegex and responseDataLabel will be empty list, hence not validating user's workload again GET attachedWorkload response in IF block.
+		//In case of responseDataWorkload, responseDataRegex empty list not going to validate user's workload again GET attachedWorkload response in IF block.
+		if len(responseDataWorkload) == 0 && len(responseDataRegex) == 0 {
+			workloads = append(workloads, workload)
+		} else {
+			if v, ok := attr[string(ocean_right_sizing_rule.WorkloadName)].(string); ok && v != "" {
+
+				// Creating list[map] so that we can check whether workload requested by user is already attached on not.
+				var list []map[string]interface{}
+				for _, obj := range responseDataWorkload {
+					item := map[string]interface{}{
+						"Name":         spotinst.StringValue(obj.Name),
+						"WorkloadType": spotinst.StringValue(obj.Type),
+						"Namespace":    spotinst.StringValue(obj.Namespace),
+					}
+					list = append(list, item)
+				}
+				userWorkload := map[string]interface{}{
+					"Name":         spotinst.StringValue(workload.Name),
+					"WorkloadType": spotinst.StringValue(workload.WorkloadType),
+					"Namespace":    spotinst.StringValue(namespaceName),
+				}
+				alreadyExist := false
+				for _, item := range list {
+					if reflect.DeepEqual(item, userWorkload) {
+						alreadyExist = true
+						break
+					}
+				}
+				if alreadyExist {
+					break
+				} else {
+					workloads = append(workloads, workload)
+				}
+			}
+			if v, ok := attr[string(ocean_right_sizing_rule.RegexName)].(string); ok && v != "" {
+				// Creating list[map] so that we can check whether workload requested by user is already attached on not.
```

Comment:
modify the statement to read "workload by regex requested

Reply:
Updated to : // Creating list[map] so that we can check whether `workload with regex` requested by user is already attached on not.